middleware/logger_test.go

```diff
@@ -0,0 +1,38 @@
+package middleware
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLogMiddlewareWritesExpectedLogs(t *testing.T) {
+	builder := new(strings.Builder)
+
+	logger := zerolog.New(builder)
+	loggingMiddleware := MakeLoggingMiddlewareHandler(logger)
+
+	nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+
+	middlewareChain := loggingMiddleware(nextHandler)
+
+	req, err := http.NewRequest("method", "url", nil)
+	assert.NoError(t, err)
+	w := httptest.NewRecorder()
+
+	middlewareChain.ServeHTTP(w, req)
+
+	var log struct {
+		Level   string `json:"level"`
+		Message string `json:"message"`
```

Comment:
zerologger attaches a time field, and I think it would be of value to verify it's coming back as not-nil. Something like

`assert.True(t, strings.Contains(log.Message, "time")`
`assert.Greater(t, 0, log.Time)`

Reply:
zerolog actually doesn't pass a time field with the default `zerologger.New`, but I can make it do that and test it.

Reply:
Oh, my bad. I saw 
```
//     log.Info().Msg("hello world")
//     // Output: {"time":1494567715,"level":"info","message":"hello world"}
```
in the docs and assumed it would do that. I would assume we want timestamps on our logs, personally, but that's not necessarily true?

Reply:
I would basically always like a timestamp, yes. 👍